.circleci/config.yml

```diff
@@ -22,7 +22,7 @@ jobs:
       - checkout
       - run: |
           cd ../
-          git clone https://github.com/mattermost/mattermost-webapp.git
+          git clone --depth=1 --no-single-branch https://github.com/mattermost/mattermost-webapp.git
```

Comment:
Why did you add `--no-single-branch` which in reality fetches the near tip of all branches? Is it needed to have the extra information available?

Reply:
It does not fetch the tip of all branches by default. My initial PR did not have "--no-single-branch" and that failed to fetch the other branch in TE and EE repos respectively.

Here's what the man page says:

>  --depth <depth>
           Create a shallow clone with a history truncated to the specified number of commits. **Implies --single-branch unless --no-single-branch is given to fetch the histories near the tips of all branches**.

Emphasis mine.

Reply:
Yes, I think my question was confusing. What I'd like to understand is that do you actually need the extra histories near the tip of other branches on the CI?

Reply:
Because we are checking out other branches than master. `git checkout $CIRCLE_BRANCH || git checkout master`. I have not done this in other places where we don't need to check out other branches.

Reply:
:+1: